app/javascript/readingList/readingList.jsx

```diff
@@ -69,12 +70,20 @@ export class ReadingList extends Component {
     this.clearSelectedTags = clearSelectedTags.bind(this);
   }
 
-  componentDidMount() {
+  async componentDidMount() {
```

Comment:
Can you explain why some of these method needs to be async?

Reply:
so as `performInitialSearch` and `selectTag` did not run in parallel -- the former must complete before the latter starts, or else it malfunctions. They both result in updating the list of posts in the view. I could theoretically make my change as a part of  `performInitialSearch`, but reusing the existing `selectTag` methods seems a less intrusive and DRY way to go.